### PR TITLE
Masun dev

### PR DESCRIPTION
**Pull request by maggiemsft**
This change adds a few more tests in system.runtime.test  to test string trim of both implementation of single char and char array. issue discussed in #1826
Appreciate comments and review
This is also the bootcamp assignment
cc: @ellismg @sneshaf @anniepoh

### CODE REVIEW

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -1617,8 +1617,11 @@ public sealed partial class String : System.Collections.Generic.IEnumerable<char
         public string ToUpper() { return default(string); }
         public string ToUpperInvariant() { return default(string); }
         public string Trim() { return default(string); }
+        public string Trim(char c) { return default(string); }
```
Comment (justinvp) [inline src/System.Runtime/ref/System.Runtime.cs]: I mentioned this in in https://github.com/dotnet/coreclr/pull/2041/files#r44751579, but mentioning it again here: Should the parameter be named `trimChar` to be consistent with the existing `trimChars`?

**Comment by justinvp** [comment]
For reference, the issue mentioned in the description is in the coreclr repo: https://github.com/dotnet/coreclr/issues/1826

**Comment by justinvp** [comment]
@ellismg, does this need to go in the future branch?

File: src/System.Runtime/ref/System.Runtime.cs (hunk 1 above)
Comment (ellismg) [inline src/System.Runtime/ref/System.Runtime.cs]: I think that's a good idea.

File: src/System.Runtime/tests/System.Runtime.Tests.csproj [hunk 2]
```diff
@@ -111,7 +111,12 @@
   </ItemGroup>
   <ItemGroup>
     <!-- Compile tests against the System.Runtime contract, but copy our local-built implementation for testing -->
-    <ProjectReference Include="..\src\System.Runtime.CoreCLR.csproj">
+    <ProjectReference Include="..\ref\System.Runtime.csproj">  
+       <ReferenceOutputAssembly>true</ReferenceOutputAssembly>  
+       <OutputItemType>None</OutputItemType>  
+       <CopyToOutputDirectory>Never</CopyToOutputDirectory>  
+     </ProjectReference>  
+<ProjectReference Include="..\src\System.Runtime.CoreCLR.csproj">
```
Comment (ellismg) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: Spacing seems off here.

File: src/System.Runtime/tests/System/String.cs [hunk 3]
```diff
@@ -1544,6 +1544,48 @@ public static void TestTrim()
 
         s = ". Foo .".TrimEnd('.');
         Assert.Equal(". Foo ", s);
+
+        s = "".Trim('c');
```
Comment (ellismg) [inline src/System.Runtime/tests/System/String.cs]: Could we take this opportunity to clean up the tests here a little?  I think using `[Theory]` and `[InlineData]` could help make this much more readable.

Maybe split this apart into three methods?  One for each method and have cases with both a single and multiple characters to split on?

We should also continue to have some coverage for cases where an array with a single element is passed into the trim overload that takes a `char[]`

File: src/System.Runtime/ref/System.Runtime.cs (hunk 1 above)
Comment (maggiemsft) [inline src/System.Runtime/ref/System.Runtime.cs]: Agree. Will change.

File: src/System.Runtime/tests/System.Runtime.Tests.csproj (hunk 2 above)
Comment (maggiemsft) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: You are right. Will fix it, thanks

File: src/System.Runtime/tests/System/String.cs (hunk 3 above)
Comment (maggiemsft) [inline src/System.Runtime/tests/System/String.cs]: I'll try to break this up and make it more readable. Yes, I'll add the test for array with single element.Thanks

**Comment by maggiemsft** [comment]
Thank you for your comments, this PR is closed

**Pull request closed without merging**